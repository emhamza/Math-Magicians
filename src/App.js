import Calculator from './components/Calculator';
import ButtonBox from './components/ButtonBox';
import Button from './components/Button';

const btnValues = [
  [{ id: 'clear', label: 'C' }, { id: 'negate', label: '+/-' }, { id: 'percent', label: '%' }, { id: 'divide', label: 'รท' }],
  [{ id: '7', label: '7' }, { id: '8', label: '8' }, { id: '9', label: '9' }, { id: 'multiply', label: 'x' }],
  [{ id: '4', label: '4' }, { id: '5', label: '5' }, { id: '6', label: '6' }, { id: 'subtract', label: '-' }],
  [{ id: '1', label: '1' }, { id: '2', label: '2' }, { id: '3', label: '3' }, { id: 'add', label: '+' }],
  [{ id: '0', label: '0' }, { id: '.', label: '.' }, { id: 'equals', label: '=' }],
];

function App() {
  return (
    <div>
      <Calculator>
        <ButtonBox>
          {btnValues.flat().map((btn) => (
            <Button
              value={btn.label}
              key={btn.id}
            />
          ))}
        </ButtonBox>
      </Calculator>
    </div>
  );
}

export default App;
