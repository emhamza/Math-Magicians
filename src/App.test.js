import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import App from './App';

describe('App', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <Router>
        <App />
      </Router>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
