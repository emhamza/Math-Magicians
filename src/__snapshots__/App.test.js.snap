// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="myHeader"
    style="display: flex; margin-top: 0px; color: white;"
  >
    <h2>
      Math Magician
    </h2>
    <ul>
      <li>
        <a
          href="/"
          style="color: white;"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          style="color: white;"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quotes"
          style="color: white;"
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
  <div
    class="content"
  >
    <h1>
      Welcome to our Home Page!
    </h1>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit, quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos sapiente officiis modi at sunt excepturi expedita sint? Sed quibusdam recusandae alias error harum maxime adipisci amet laborum. Perspiciatis minima nesciunt dolorem! Officiis iure rerum voluptates a cumque velit quibusdam sed amet tempora. Sit laborum ab, eius fugit doloribus tenetur fugiat, temporibus enim commodi iusto libero magni deleniti quod quam consequuntur! Commodi minima excepturi repudiandae velit hic maxime doloremque. Quaerat provident commodi consectetur veniam similique ad earum omnis ipsum saepe, voluptas, hic voluptates pariatur est explicabo fugiat, dolorum eligendi quam cupiditate excepturi mollitia maiores labore suscipit quas? Nulla, placeat. Voluptatem quaerat non architecto ab laudantium modi minima sunt esse temporibus sint culpa, recusandae aliquam numquam totam ratione voluptas quod exercitationem fuga. Possimus quis earum veniam quasi aliquam eligendi, placeat qui corporis!
    </p>
  </div>
</DocumentFragment>
`;
